### PR TITLE
Fix part of Coverity hits

### PR DESCRIPTION
**Pull request by kboyarinov**
### Description 
Fixed "Using a moved object" bug in `concurrent_hash_map`.
Improved source code around "Uninitialized pointer fields" hits.

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [x] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: include/oneapi/tbb/concurrent_hash_map.h [hunk 1]
```diff
@@ -158,7 +158,8 @@ class hash_map_base {
         if (is_initial) {
             init_buckets_impl(ptr, sz);
         } else {
-            init_buckets_impl(ptr, sz, reinterpret_cast<node_base*>(rehash_req_flag));
+            node_base* bucket_initializer = reinterpret_cast<node_base*>(rehash_req_flag);
+            init_buckets_impl(ptr, sz, bucket_initializer);
```
Comment (pavelkumbrasev) [inline include/oneapi/tbb/concurrent_hash_map.h:162]:
```suggestion
            init_buckets_impl(ptr, sz, /* bucket_initializer = */ reinterpret_cast<node_base*>(rehash_req_flag));
```

Comment (pavelkumbrasev) [inline include/oneapi/tbb/concurrent_hash_map.h:162]:
```suggestion
            init_buckets_impl(ptr, sz, /* bucket_initializer = */ reinterpret_cast<node_base*>(rehash_req_flag));
```

File: include/oneapi/tbb/detail/_aggregator.h [hunk 2]
```diff
@@ -155,14 +155,17 @@ class aggregator : public aggregator_generic<OperationType> {
 // template<class U, class V> friend class aggregating_functor;
 template <typename AggregatingClass, typename OperationList>
 class aggregating_functor {
-    AggregatingClass* my_object;
+    AggregatingClass* my_object = nullptr;
```
Comment (pavelkumbrasev) [inline include/oneapi/tbb/detail/_aggregator.h:158]:
```suggestion
    AggregatingClass* my_object{nullptr};
```

File: include/oneapi/tbb/concurrent_hash_map.h (hunk 1 above)
Comment (kboyarinov) [reply include/oneapi/tbb/concurrent_hash_map.h:162]: `reinterpret_cast` will produce rvalue of type `node_base*`. Suggested change will not compile

**Review by pavelkumbrasev — APPROVED** [review]

**Pull request merged**